alacritty/src/display/content.rs

```diff
@@ -222,18 +222,22 @@ impl RenderableCell {
         let viewport_start = Point::new(Line(-(display_offset as i32)), Column(0));
         let colors = &content.config.colors;
         let mut character = cell.c;
-
-        if let Some((c, is_first)) =
+        let mut flags = cell.flags;
+        if let Some(hint_type) =
             content.hint.as_mut().and_then(|hint| hint.advance(viewport_start, cell.point))
         {
-            let (config_fg, config_bg) = if is_first {
-                (colors.hints.start.foreground, colors.hints.start.background)
-            } else {
-                (colors.hints.end.foreground, colors.hints.end.background)
-            };
-            Self::compute_cell_rgb(&mut fg, &mut bg, &mut bg_alpha, config_fg, config_bg);
-
-            character = c;
+            match hint_type {
+                HintType::Label(c, is_first) => {
+                    let (config_fg, config_bg) = if is_first {
+                        (colors.hints.start.foreground, colors.hints.start.background)
+                    } else {
+                        (colors.hints.end.foreground, colors.hints.end.background)
+                    };
+                    Self::compute_cell_rgb(&mut fg, &mut bg, &mut bg_alpha, config_fg, config_bg);
+                    character = c;
+                },
+                HintType::AfterLabel => flags |= Flags::UNDERLINE,
+            }
```

Comment:
I think on this own the approach taken here isn't bad, the problem really is just that this method is an incredibly ugly behemoth that is slowly becoming unmaintainable.

I don't think there's anything simple you can do about this though, so I'd probably just accept the additional complexity for this PR.

Reply:
Sorry of the late reply, I agree, probably can't be solved as easily as I thought.
I don't really like this solution as well, but can't come up with a cleaner way, so I think I will just close the pr.